docs/platforms/javascript/common/troubleshooting/index.mdx

```diff
@@ -221,7 +221,8 @@ const SENTRY_PROJECT_IDS = ["123456"];
 
 export const postAction = async ({ request }) => {
   try {
-    const envelope = await request.text();
+    const envelopeBytes = await req.arrayBuffer();
+    const envelope = new TextDecoder().decode(envelopeBytes);
```

Comment:
Looks like `TextDecoder` anyway[ expects UTF-8 or other type string](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder). What's the advantage from calling `.text` on request?

Reply:
There are 2 explanations:

- Session replays are sent as compressed zip (binary) appended at the end of the envelope body. When you get the content of the session replay as text you lose data in the transfer
- if you use `.text()` then the body was used and you can't reuse it from the request to get its bytes

Therefore I get the bytes and save it separately to forward to Sentry and then I convert the bytes to text to extract the textual part of it to validate.

This is already validated and running in production for me. Works very well.